app/Domain/Timesheets/Repositories/Timesheets.php

```diff
@@ -276,37 +277,36 @@ public function getWeeklyTimesheets(int $projectId, Carbon $fromDate, int $userI
             zp_projects.name,
             zp_projects.id AS projectId,
             zp_projects.clientId AS clientId,
-            zp_clients.name AS clientName,
-            GROUP_CONCAT(zp_timesheets.workDate SEPARATOR ',') as workDates
-            , ROUND(sum(case when TO_DAYS(zp_timesheets.workDate) = TO_DAYS(:dateStart1) then zp_timesheets.hours else 0 end),2) as hoursMonday
-            , ROUND(sum(case when TO_DAYS(zp_timesheets.workDate) = (TO_DAYS(:dateStart1) + 1) then zp_timesheets.hours else 0 end),2) as hoursTuesday
-            , ROUND(sum(case when TO_DAYS(zp_timesheets.workDate) = (TO_DAYS(:dateStart1) + 2) then zp_timesheets.hours else 0 end),2) as hoursWednesday
-            , ROUND(sum(case when TO_DAYS(zp_timesheets.workDate) = (TO_DAYS(:dateStart1) + 3) then zp_timesheets.hours else 0 end),2) as hoursThursday
-            , ROUND(sum(case when TO_DAYS(zp_timesheets.workDate) = (TO_DAYS(:dateStart1) + 4) then zp_timesheets.hours else 0 end),2) as hoursFriday
-            , ROUND(sum(case when TO_DAYS(zp_timesheets.workDate) = (TO_DAYS(:dateStart1) + 5) then zp_timesheets.hours else 0 end),2) as hoursSaturday
-            , ROUND(sum(case when TO_DAYS(zp_timesheets.workDate) = (TO_DAYS(:dateStart1) + 6) then zp_timesheets.hours else 0 end),2) as hoursSunday
+            zp_clients.name AS clientName
         FROM
             zp_timesheets
         LEFT JOIN zp_tickets ON zp_tickets.id = zp_timesheets.ticketId
         LEFT JOIN zp_projects ON zp_tickets.projectId = zp_projects.id
         LEFT JOIN zp_clients ON zp_clients.id = zp_projects.clientId
         WHERE
-            ((TO_DAYS(zp_timesheets.workDate) >= TO_DAYS(:dateStart1)) AND (TO_DAYS(zp_timesheets.workDate) < (TO_DAYS(:dateStart2) + 7)))
+            (zp_timesheets.workDate >= :dateStart1 AND zp_timesheets.workDate < :dateEnd)
             AND (zp_timesheets.userId = :userId)
+            AND hours > 0
         ";
 
         if ($projectId > 0) {
             $query .= " AND zp_tickets.projectId = :projectId";
         }
-        $query .= "GROUP BY ticketId, kind";
+
+        $query .= " ORDER BY zp_timesheets.ticketId, zp_timesheets.kind, zp_timesheets.workDate DESC";
 
         $call = $this->dbcall(func_get_args());
 
         $call->prepare($query);
 
-        $startOfDay = $fromDate->format('Y-m-d');
+        if (!$fromDate->isUtc()) {
+            $fromDate->setTimezone("UTC");
+        }
+        $startOfDay = $fromDate->format('Y-m-d H:i:s');
```

Comment:
If possible I think we should move the "format" into a const (to help prevent typos). Could be DateTimeHelper::DatabaseTimeFormat or something like that.

Reply:
Yeah I agree. Actually, I moved to using the helper methods and not having to worry about the format at all. 
It's either dbDateTime (UTC + iso format) or userDateTime (local timezone + user format). There is one exception where I need to be able to handle 24:00 time for input type="time" which requires it.